feat(packages): add run-loops package for autonomous agent operation

TimeToBuildBob opened the pull request:
## Summary

Adds the run-loops Python package providing infrastructure for autonomous AI agent operation.

## Features

- **Autonomous run loops**: Base framework for scheduled/triggered agent execution with systemd timer integration
- **Project monitoring**: GitHub PR/issue monitoring with automated responses, CI failure handling
- **Email integration**: Email-based communication loops with mbsync support
- **Utilities**: Distributed locking, structured logging, GitHub API wrapper, git operations

## Motivation

This package enables agents like Alice to set up autonomous operation by using shared infrastructure from gptme-contrib, rather than each agent maintaining separate implementations.

Currently Bob has this package in his workspace, but Alice (and other agents) need access to run autonomously. By upstreaming to gptme-contrib, all agents can:
1. Use the same run loop infrastructure
2. Benefit from improvements made by any agent
3. Maintain consistent autonomous operation patterns

## Components

| Module | Purpose |
|--------|---------|
| `autonomous.py` | Main autonomous operation loop |
| `project_monitoring.py` | GitHub monitoring and automated responses |
| `email.py` | Email-based communication |
| `base.py` | Base class for run loops |
| `utils/` | Locking, logging, GitHub, git utilities |

## Usage

```bash
# Install
uv pip install -e packages/run_loops

# CLI
run-loops autonomous --workspace /path/to/workspace
run-loops project-monitoring --workspace /path/to/workspace
run-loops email --workspace /path/to/workspace
```

## Tests

Includes test suite covering:
- Base run loop functionality
- Autonomous runner
- Project monitoring
- Email integration
- Utility functions (locking, GitHub)

## Related

- ErikBjare/bob#193 - Getting inter-agent communication working with Alice

/cc @ErikBjare
<!-- ELLIPSIS_HIDDEN -->


----

> [!IMPORTANT]
> Adds `run-loops` package for autonomous AI agent operations with CLI, Python API, utilities, and comprehensive tests.
> 
>   - **Features**:
>     - Adds `run-loops` package for autonomous AI agent operations with modules `autonomous.py`, `project_monitoring.py`, `email.py`, and `base.py`.
>     - CLI commands for `autonomous`, `project-monitoring`, and `email` operations in `cli.py`.
>     - Utilities for locking, logging, GitHub API, and git operations in `utils/`.
>   - **Tests**:
>     - Comprehensive test suite for all modules, including `test_autonomous.py`, `test_base.py`, `test_email.py`, `test_project_monitoring.py`, and `test_utils/test_github.py`.
>     - Integration tests for CLI commands in `test_integration.py`.
>   - **Misc**:
>     - Updates `.pre-commit-config.yaml` to exclude `run_loops` from mypy checks.
>     - Adds `README.md` and `pyproject.toml` for package documentation and configuration.
> 
> <sup>This description was created by </sup>[<img alt="Ellipsis" src="https://img.shields.io/badge/Ellipsis-blue?color=175173">](https://www.ellipsis.dev?ref=gptme%2Fgptme-contrib&utm_source=github&utm_medium=referral)<sup> for 27ec07d745a81b3d61745822a605eb7b53e4c81e. You can [customize](https://app.ellipsis.dev/gptme/settings/summaries) this summary. It will automatically update as commits are pushed.</sup>


<!-- ELLIPSIS_HIDDEN -->

ErikBjare commented on packages/run_loops/src/run_loops/utils/execution.py:
```diff
@@ -0,0 +1,120 @@
+        # Build gptme command
+        # Make sure we are using the system (pipx-managed) gptme and not the one in the uv-workspace
+        # TODO: do this cleaner by modifying the PATH?
+        cmd = ["/home/bob/.local/bin/gptme"]
```
Yes, although it would also mean that it runs the gptme installed in whatever uv environment the run_loops runs in, not the user installed stable gptme that we wanted to use for the core loop for stability reasons (in case something happens to the gptme in the uv env, outdated or unstable etc).

TimeToBuildBob commented:
## Addressed Review Feedback

### Shell Safety & Log Handling Fixes (commit 999452d)
- ✅ **Quote log_file path**: Prevents shell injection with paths containing spaces
- ✅ **Use append mode**: Timeout handler now appends instead of overwriting, preserving tee output

### Erik's Question on mypy Exclusion

Good question! The packages are excluded from the top-level mypy pre-commit because:

1. **Workspace typing conflicts**: Mypy has issues running across uv workspaces with different package roots. The pre-commit runs from repo root, but packages need their own mypy path configuration.

2. **Each package can have its own mypy config**: Individual packages can define their own mypy settings in pyproject.toml with proper mypy_path.

Your suggestion of a separate `make typecheck` target makes sense. I can implement:
- A top-level `make typecheck` that iterates through packages
- Each package has its own `make typecheck` target  
- Pre-commit calls `make typecheck`

Let me know if you'd like this in a follow-up commit or if we should address it separately.

ErikBjare commented:
I assume tests are not being run for packages in CI yet? They should!

TimeToBuildBob commented on .pre-commit-config.yaml:
```diff
@@ -37,7 +37,7 @@ repos:
     types: [python]
     pass_filenames: true
-    exclude: ^packages/lessons/
+    exclude: ^packages/(lessons|run_loops)/
```
✅ Addressed in latest push:

1. Added Makefile with `typecheck` target that runs mypy on each package
2. Added `typecheck-packages` pre-commit hook that calls `make typecheck`
3. Kept mypy pre-commit for scripts, Makefile handles packages

This ensures all code is typechecked - scripts via pre-commit mypy, packages via make typecheck.

ErikBjare replied:
Oh, you did? But I merged the PR? Maybe it didn't get included?
